### PR TITLE
docs: explain stable links in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by jnewbery**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -34,6 +34,27 @@ Some tips for hosting a great review club meeting:
...
   own research from there.
 
+- When linking to code, use stable links that won't break later:
+
+  - for linking to code in the master branch, link to a recent commit hash in
+    the master branch of bitcoin/bitcoin, e.g.
+    [https://github.com/bitcoin/bitcoin/blob/23d3ae7a/src/addrman.cpp#L44](https://github.com/bitcoin/bitcoin/blob/23d3ae7a/src/addrman.cpp#L44).
+    You can use a short (8 character) commit id. REASON: linking directly to master
+    isn't stable because line numbers/files change over time as the HEAD commit of
+    master changes.
```
Comment (jonatack) [inline CONTRIBUTING.md:44]: perhaps omit "as the HEAD commit of master changes" or just "as the files change over time."

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -34,6 +34,27 @@ Some tips for hosting a great review club meeting:
...
+  - for linking to code in the PR branch, ask one of the
+    bitcoin-core-review-club maintainers to add a tag of the current PR branch
+    to bitcoin-core-pr-reviews/bitcoin, and link to that branch or a commit on
```
Comment (jonatack) [inline CONTRIBUTING.md:47]: `s/bitcoin-core-pr-reviews/bitcoin-core-review-club/`

maybe add the link: [bitcoin-core-review-club/bitcoin](https://github.com/bitcoin-core-review-club/bitcoin) here and in line 55

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -34,6 +34,27 @@ Some tips for hosting a great review club meeting:
...
+- When linking to code, use stable links that won't break later:
+
+  - for linking to code in the master branch, link to a recent commit hash in
+    the master branch of bitcoin/bitcoin, e.g.
+    [https://github.com/bitcoin/bitcoin/blob/23d3ae7a/src/addrman.cpp#L44](https://github.com/bitcoin/bitcoin/blob/23d3ae7a/src/addrman.cpp#L44).
+    You can use a short (8 character) commit id. REASON: linking directly to master
+    isn't stable because line numbers/files change over time as the HEAD commit of
+    master changes.
+  - for linking to code in the PR branch, ask one of the
+    bitcoin-core-review-club maintainers to add a tag of the current PR branch
```
Comment (jonatack) [inline CONTRIBUTING.md:46]: might be easier to read as s/bitcoin-core-review-club/review club/ maintainers

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -34,6 +34,27 @@ Some tips for hosting a great review club meeting:
...
+    to bitcoin-core-pr-reviews/bitcoin, and link to that branch or a commit on
+    that branch, e.g.
+    [https://github.com/bitcoin-core-review-club/bitcoin/commit/a6ca5080#diff-be2905e2f5218ecdbe4e55637dac75f3R1751-R1754](https://github.com/bitcoin-core-review-club/bitcoin/commit/a6ca5080#diff-be2905e2f5218ecdbe4e55637dac75f3R1751-R1754).
+    You can use a short (8 character) commit id. REASON: if the PR auther modifies
```
Comment (jonatack) [inline CONTRIBUTING.md:50]: s/auther/author/

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -34,6 +34,27 @@ Some tips for hosting a great review club meeting:
...
+    their branch, links to the PR branch will become invalidated. If they
+    force-push and the commits are detached, those commits will eventually get
+    garbage-collected by GitHub.
```
Comment (jonatack) [inline CONTRIBUTING.md:53]: maybe s/get garbage collected/be removed/

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -34,6 +34,27 @@ Some tips for hosting a great review club meeting:
...
+
+for code in the PR: the commit hash in the PR branch of bitcoin-core-review-club/bitcoin
+for code in master: a (recent) commit hash in the master branch of bitcoin/bitcoin
```
Comment (jonatack) [inline CONTRIBUTING.md:56]: For both of these two lines:

- missing preceding hyphens to be in the list

- s/: /: link to/

- missing full stop

**Review by jonatack — COMMENTED** [review]
Thanks John! A some suggestions below. Also, in the rest of this doc, bullet points begin as capitalised sentences.

File: CONTRIBUTING.md (hunk 6 above)
Comment (jonatack) [reply CONTRIBUTING.md:56]: Oh, are these two lines preliminary notes that should be removed?

**Comment by jnewbery** [comment]
Thanks for the review @jonatack (and apologies for the very sloppy first draft - I did this in a hurry and wasn't careful in checking grammar/spelling).

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -34,6 +34,23 @@ Some tips for hosting a great review club meeting:
...
 
+- When linking to code, use stable links that won't break later:
+
+  - for linking to code in the master branch, link to a recent commit hash in
+    the master branch of bitcoin/bitcoin, e.g.
+    [https://github.com/bitcoin/bitcoin/blob/23d3ae7a/src/addrman.cpp#L44](https://github.com/bitcoin/bitcoin/blob/23d3ae7a/src/addrman.cpp#L44).
+    You can use a short (8 character) commit id. REASON: linking directly to master
+    isn't stable because line numbers/files change over time.
+  - for linking to code in the PR branch, ask one of the
+    Review Club maintainers to add a tag of the current PR branch
```
Comment (jonatack) [inline CONTRIBUTING.md:45]: oooh *capitalized* review club! (it's lowercased most of the time)

**Comment by jonatack** [comment]
LGTM, thanks for the helpful info, John

**Comment by jnewbery** [comment]
Thanks Jon. Decapitalized ReViEw ClUb and merged.

**Pull request merged**